pkg/check/contentavailability/contentavailability.go

```diff
@@ -0,0 +1,178 @@
+	if err != nil {
+		return err
+	}
+	client := clients[node]
+
+	overlays, err := cluster.FlattenOverlays(ctx)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("%s: %s\n", node, overlays[node])
+
+	batchID, err := client.GetOrCreateBatch(ctx, o.PostageAmount, o.PostageDepth, o.GasPrice, o.PostageLabel)
+	if err != nil {
+		return fmt.Errorf("node %s: unable to create batch id: %w", node, err)
+	}
+	fmt.Printf("node %s: batch id %s\n", node, batchID)
+
+	time.Sleep(5 * time.Second) // Wait for nodes to get notified of the batch..
+
+	contentAddr, err := client.UploadBytes(ctx, content, api.UploadOptions{BatchID: batchID})
+	if err != nil {
+		return fmt.Errorf("node %s: unable to upload content: %w", node, err)
+	}
+	fmt.Printf("node %s: content uploaded successfully: %s\n", node, addr)
+
+	time.Sleep(5 * time.Second) // Wait for nodes to sync.
+
+	isRetrievable, err := client.IsRetrievable(ctx, contentAddr)
+	if err != nil {
+		return fmt.Errorf("node %s: unable to check if content is retrievable: %w", node, err)
+	}
+	if !isRetrievable {
+		return fmt.Errorf("node %s: the uploaded content is not retrievable", node)
+	}
+	fmt.Printf("node %s: uploaded content is retrievable\n", node)
+
+	addresses := store.Addresses()
+	if len(addresses) == 0 {
+		return errors.New("empty list of addresses")
+	}
+
+	rmCount := 0
+	rmChAddr := addresses[len(addresses)-1]
+	for node, nClient := range clients {
+		if err := nClient.RemoveChunk(ctx, rmChAddr); err != nil {
+			return fmt.Errorf("node %s: unable to remove chunk %s: %w", node, rmChAddr, err)
+		}
+		fmt.Printf("node %s: chunk %s removed\n", node, rmChAddr)
+
+		if rmCount++; rmCount == len(clients) {
+			break
+		}
+
+		time.Sleep(5 * time.Second) // Wait for nodes to sync.
+
+		isRetrievable, err = client.IsRetrievable(ctx, contentAddr)
+		if err != nil {
+			return fmt.Errorf("node %s: unable to check if content is retrievable: %w", node, err)
+		}
+		if !isRetrievable {
+			return fmt.Errorf("node %s: the uploaded content is not retrievable", node)
+		}
+		fmt.Printf("node %s: the uploaded content is retrievable\n", node)
+	}
+	isRetrievable, err = client.IsRetrievable(ctx, contentAddr)
+	if err != nil {
+		return fmt.Errorf("node %s: unable to check if content is retrievable: %w", node, err)
+	}
+	if isRetrievable {
+		return fmt.Errorf("node %s: the uploaded content is retrievable", node)
+	}
+	fmt.Printf("node %s: the uploaded content is not retrievable\n", node)
+
+	return nil
+}
+
+type putterMock struct {
+	chunks []swarm.Chunk
+	exists map[string]swarm.Address
+}
+
+func (pm *putterMock) Put(_ context.Context, _ storage.ModePut, chs ...swarm.Chunk) ([]bool, error) {
+	exists := make([]bool, len(chs))
+	for i, ch := range chs {
+		key := ch.Address().ByteString()
+		if _, ok := pm.exists[key]; ok {
+			exists[i] = true
+			continue
+		}
+		pm.chunks = append(pm.chunks, ch)
+		pm.exists[key] = ch.Address()
+	}
+	return exists, nil
+}
+
+func (pm *putterMock) Chunks() []swarm.Chunk {
```

Comment:
can't find where this is used. I think that since you don't really care about the actual chunk data once `Put` is called, you could get away with just having a map of `map[string]struct{}` that would just hold the chunk addresses

Reply:
Removed.